lib/haskell/anyall/src/AnyAll/Types.hs

```diff
@@ -95,6 +97,10 @@ instance Semigroup t => Semigroup (Label t) where
   Pre pr1 <> PrePost pr2 po2 = PrePost (pr1 <> pr2) po2
   PrePost pr1 po1 <> Pre pr2 = PrePost (pr1 <> pr2) po1
   PrePost pr1 po1 <> PrePost pr2 po2 = PrePost (pr1 <> pr2) (po1 <> po2)
+  -- doesn't really matter, nothing is parsed into Metadata /IL
+  Metadata x <> Metadata y = Metadata (x <> y)
+  Metadata _ <> x = x
```

Comment:
This may be a foot gun in the future. In theory, this case should never happen, but what if? Perhaps it would be better to throw an error in this case?